x-pack/plugins/alerting/server/alert/create_alert_factory.ts

```diff
@@ -140,6 +140,7 @@ export function createAlertFactory<
             previouslyRecoveredAlerts: {},
             hasReachedAlertLimit,
             alertLimit: maxAlerts,
+            autoRecoverAlerts: true,
```

Comment:
shouldn't this use the value as defined by the rule type?

Reply:
Oh okay, I wasn't sure if it should be set by the rule types because it's just used to return the recovered alerts and is not used in the same way as the task runner but that makes sense.

Reply:
Resolved in this commit 9d7056ee03ffaa203f6891689e5aefeb8ffe3ab2